### PR TITLE
FFmpeg mostly working in linux

### PR DESCRIPTION
**Pull request by Sebanisu**
Not Working Video:
Frame image extraction to Texture2D
I think it might be a codec issue. maybe. I was getting a CTD when trying to free the memory to a AVCodec* could be related to the lack of video. I will work on figuring out why it doesn't work text week.
I also wanted to bypass the sws scaler to see if that fixed it.

Working:
Video audio is playing back 
OGG music through ffmpeg

The audio from ffmpeg is being loaded a bit at a time controlled by a variable. Instead of loading the entire thing into memory then playing it back. If you experience audio skipping then you may need to increase the number. The buffer plays it back and it skips when it runs out of stuff to playback. It's similar to how video loads one frame at a time. Audio is more like loads 75 frames and tries to keep that number up there.

I got sick this week so I had trouble making much progress.

I had spent most of my time trying to find memory leaks.

The Ffmpeg.Autogen.dll is custom compiled by me as it had a bug in it that prevented it from working on linux. Unless our target was .net 2.0. The bug will be fixed in a new version. 

I had to compile in 64bit to use ffmpeg on my linux distro. I included both of them for windows incase we got around to removing that old midi lib. 

Might need to customize linux library detection depending on where the files are located for people. FFmpegBinariesHelper.cs has a line 34 
libraryPath = "/usr/lib/x86_64-linux-gnu";
I think that is what was working for me.  I also have the ffmpeg.autogen.dll.config. but i don't think that is doing anything.

oh yeah I tweaked the debug menu for playing music to show you what song you are on. arrow keys change enter will play and stop.

### CODE REVIEW

File: FF8/Memory.cs [hunk 1]
```diff
@@ -28,20 +25,64 @@ class Memory
...
         public static void SpriteBatchStartStencil()
-        => spriteBatch.Begin(SpriteSortMode.BackToFront, BlendState.Opaque, null, graphics.GraphicsDevice.DepthStencilState);
+        {
+            spriteBatch.Begin(SpriteSortMode.BackToFront, BlendState.Opaque, null, graphics.GraphicsDevice.DepthStencilState);
+        }
+
         public static void SpriteBatchStartAlpha()
-            => spriteBatch.Begin(SpriteSortMode.Deferred, BlendState.AlphaBlend);
+        {
+            spriteBatch.Begin(SpriteSortMode.Deferred, BlendState.AlphaBlend);
```
Comment (MaKiPL) [inline FF8/Memory.cs]: is there any specific reason for changing lambda expression to method block? Readability or something else?

**Comment by MaKiPL** [comment]
Merged, thank you kindly

File: FF8/Memory.cs (hunk 1 above)
Comment (Sebanisu) [reply FF8/Memory.cs]: I had setup format document in visual studio. I try to run that on the document i'm working on to fix the tabs and such. That is one of the things it's set to do right now. I can turn it off I think.

Comment (Sebanisu) [reply FF8/Memory.cs]: Now that I think about it I might of done this manually. And I completely forgot about it. I'll change it back.

**Pull request merged**